src/cassette.jl

```diff
@@ -141,6 +143,143 @@ function replacefunc(f::Function, d::AbstractDict)
     ctx = GraftCtx(metadata=d)
     return Cassette.recurse(ctx, f)
 end
+            
+#------------------------------------------------------------------------------------           
+            
+Cassette.@context TypeCtx
+            
+"""   TypeCtx
+
+creates a MetaDiGraph tracking the types of args and ret values throughout a script
+
+"""
+TypeCtx
+            
+            
+
+"""   FCollector(depth::Int,frame::function,data::FCollector)
+
+struct to collect all the functions called throughout a script
+        
+"""
+mutable struct FCollector{I,F,C}
+    depth::I
+    frame::F
+    data::Vector{C}
+end
+            
+
+
+"""   FCollector(depth::Int,frame::Frame)
+
+this is an initialization funtion for the FCollector
+
+"""
+function FCollector(d::Int, f)
+    FCollector(d, f, FCollector[])
+end
+
+""" Frame(func, args, ret, subtrace)
+
+a structure to hold metadata for recursive type information for every function called
```

Comment:
connect the idea of a function call `Frame` to the `F` in `FCollector`

Reply:
and to a stack frame.

Reply:
Is this better?
![image](https://user-images.githubusercontent.com/41996797/53338204-1f8a7a00-38d1-11e9-8eb2-a968dc0a7e8f.png)

Reply:
yeah